### PR TITLE
Implement 'create emoji ...'

### PR DESCRIPTION
**Pull request by asottile**
sorry about the ram

```
(3:47:35 PM) asottile: create-asottile: emoji cat
(3:47:35 PM) create-asottile: asottile: Showing 1-50 of 82 results
(3:47:35 PM) create-asottile: asottile: ªº×ʕ໋࡙༁၌⊍⌖⑆✆✕✖⨰⨱⨴⨵⨶⨷⨻⫁⫂⯐〾ꊶꧦꩰ챁𐇬𐩿𝅼𝅾𝅿𝆀𝆁𝆭𝩿𝪀𝪁𝪂𝪃𝪄𝪅𝪆🇦🇧🇨🇩🇪
(3:47:43 PM) asottile: lol
(3:47:58 PM) asottile: create-asottile: emoji thinking
(3:47:58 PM) create-asottile: asottile: 🤔
(3:48:05 PM) asottile: create-asottile: remoji 🤔
(3:48:05 PM) create-asottile: asottile: 🤔: THINKING FACE
(3:48:12 PM) asottile: create-asottile: emoji penguin
(3:48:12 PM) create-asottile: asottile: 🐧
(3:48:17 PM) asottile: create-asottile: remoji 🐧
(3:48:17 PM) create-asottile: asottile: 🐧: PENGUIN
(3:50:19 PM) asottile: create-asottile: emoji aoisjdfkljasdf
(3:50:20 PM) create-asottile: asottile: No results 😢
```

### CODE REVIEW

File: ircbot/emoji.py [hunk 1]
```diff
@@ -1,5 +1,32 @@
 import unicodedata
 
+emoji_mapping = []
+for i in range(0x10ffff):
```
Comment (jvperrin) [inline ircbot/emoji.py]: Well that's one way to do it lol

File: ircbot/emoji.py [hunk 2]
```diff
@@ -1,6 +1,34 @@
...
 
 
+char_mapping = []
+for i in range(0x10ffff):
+    c = chr(i)
+    try:
+        name = unicodedata.name(c)
+    except ValueError:
+        continue
+    if unicodedata.category(c).startswith('C'):
```
Comment (jvperrin) [inline ircbot/emoji.py]: Why is this here? I think a comment about why this in particular is being skipped would be good.

File: ircbot/emoji.py [hunk 3]
```diff
@@ -1,6 +1,34 @@
...
+
+def emoji(respond, query):
+    query = query.upper()
+    ret = ''
+    for name, c in char_mapping:
+        if query in name:
+            ret += c
+    if not ret:
+        respond('No results 😢')
+    elif len(ret) > 50:
```
Comment (jvperrin) [inline ircbot/emoji.py]: How many lines would 50 results be? Maybe like 10 or something would be better to have less spam? Idk

Comment (asottile) [inline ircbot/emoji.py]: it puts all the characters on one line, see the example in the PR

File: ircbot/emoji.py (hunk 2 above)
Comment (asottile) [inline ircbot/emoji.py]: **C**ontrol

File: ircbot/emoji.py (hunk 3 above)
Comment (jvperrin) [inline ircbot/emoji.py]: Ah ok, then this is fine

File: ircbot/emoji.py (hunk 2 above)
Comment (jvperrin) [inline ircbot/emoji.py]: Ok, so like `# Skip control characters, because nobody wants to print those`, sounds good.

Comment (jvperrin) [inline ircbot/emoji.py]: Meh, alright, it just wasn't clear when I read through it, but I also don't have much experience with unicode.
